tests/test_search_parameters.py

```diff
@@ -74,6 +74,14 @@ def setUp(self):
             max_range=10,
         )
 
+        self.search_parameters._add_search_parameter(
```

Comment:
Concurrency and request rate will never be in the same configuration. This is fine for unit testing, but if we're going to do this we need a lot a comments explaining why. Probably better to just create a new search_parameters instance that has request rate instead of concurrency.